src/reflect/gclayout_none.go

```diff
@@ -0,0 +1,9 @@
+//go:build !gc.precise
+
+package reflect
+
+import "unsafe"
+
+func (r *rawType) gcLayout() unsafe.Pointer {
+    return nil
+}
```

Comment:
I think it's fine to pass the real `gcLayout` in all cases. When not using the precise GC, the parameter is ignored. I think the optimizer will be smart enough to optimize it all away. (And honestly, I think the precise GC is the way forward anyway on most platforms).